### PR TITLE
Support for new link parsing + syntax

### PR DESCRIPTION
**Pull request by vhyrro**
This PR adds support for a few things:
- [x] Add support for new link parsing available in the `attached-modifiers` branch of the treesitter parser
- [x] Basic queries for highlighting attached modifiers
- [x] "Won't do" syntax for links `- [-] Something I won't do`
- [x] Completely rewrite enter link on \<CR\> functionality. Will allow for more efficient and less janky code in general.
    - [x] Support new link syntax that disallows several file targets and only allows for one
    - [x] Support `@` location tag (`[link](@non-norg-file.txt)`)
    - [x] Support file-only locations (`[link](:some-norg-file:)`

### CODE REVIEW

**Comment by vhyrro** [comment]
new-highlights

This is just a showcase of the new highlights that will be available for links :D
Much richer than the basic regex highlights we had earlier.

**Comment by vhyrro** [comment]
Turns out there'll be more to this PR than we may have expected. We're thinking of doing a complete overhaul to the syntax of links and as such this code may grow obsolete fast 😳. I think the way we'll do this is I'll implement this, merge it, then we'll change the code to work with the new link syntax as well in a new PR.

We'll be showing our syntactical adjustments in the github issues in the following days/weeks. Stay tuned!

File: README.md [hunk 1]
```diff
@@ -288,6 +288,31 @@ sources = {
...
+end
+
+-- If Neorg is loaded already then don't hesitate and load the completion
+if neorg.is_loaded() then
+    load_completion()
+else -- Otherwise wait until Neorg gets started and load the completion module then
+    neorg.callbacks.on_event("core.started", load_completion)
+end
+```
+
```
Comment (mrossinek) [inline README.md:315]: Why is this here? Was this committed to the wrong branch? If so, shall we extract this and put it onto `unstable`?

File: README.md [hunk 2]
```diff
@@ -386,6 +411,7 @@ heartwarming and fuels the urge to keep going :heart:. You can support me here:
 - [Donate directly via paypal](https://paypal.me/ewaczupryna?locale.x=en_GB)
 - [Support me on Patreon](https://patreon.com/vhyrro)
 - Donate to my monero wallet: `86CXbnPLa14F458FRQFe26PRfffZTZDbUeb4NzYiHDtzcyaoMnfq1TqVU1EiBFrbKqGshFomDzxWzYX2kMvezcNu9TaKd9t`
+- Donate via bitcoin: `bc1q4ey43t9hhstzdqh8kqcllxwnqlx9lfxqqh439s`
```
Comment (mrossinek) [inline README.md:414]: Same with this

File: doc/breaking-changes.norg [hunk 3]
```diff
@@ -95,5 +95,28 @@
...
+      #a comment inbetween#
+      and some text below.
+      @end
+
+  *** New
+      @code norg
+      This is some text
+      +a comment inbetween+
+      and some text below.
+      @end
```
Comment (mrossinek) [inline doc/breaking-changes.norg:119]: This is already the case on `unstable`. Arguably this should also be picked onto that branch.

File: lua/neorg/modules/core/highlights/module.lua [hunk 4]
```diff
@@ -37,12 +37,13 @@ module.config.public = {
...
+            Verbatim = {
                 reference = "Normal",
                 percentage = 20,
             },
-            Comment = {
+
+            InlineComment = {
                 reference = "Normal",
                 percentage = 40,
             },
```
Comment (mrossinek) [inline lua/neorg/modules/core/highlights/module.lua:49]: This is currently in conflict with my proposal over in #117. What is your reasoning for using dimming rather than e.g. `+TSComment` for inline comments?

File: lua/neorg/modules/core/integrations/treesitter/module.lua [hunk 5]
```diff
@@ -261,6 +267,79 @@ module.config.public = {
             Parameters = "+TSComment",
         },
 
+        Link = {
+            Text = {
+                [""] = "+TSURI",
+                Delimiter = "+mkdDelimiter",
```
Comment (mrossinek) [inline lua/neorg/modules/core/integrations/treesitter/module.lua:273]: I don't think this is a standard highlight group, is it?

File: lua/neorg/modules/core/integrations/treesitter/module.lua [hunk 6]
```diff
@@ -261,6 +267,79 @@ module.config.public = {
...
+                    },
+                },
+            },
+        },
+
+        Markup = {
+            Bold = "cterm=bold gui=bold",
+            Italic = "cterm=italic gui=italic",
+            Strikethrough = "cterm=strikethrough gui=strikethrough",
+            Underline = "cterm=underline gui=underline",
```
Comment (mrossinek) [inline lua/neorg/modules/core/integrations/treesitter/module.lua:339]: Why aren't you using `+TSStrong` etc. for these? See e.g. my proposal over in #117

File: lua/neorg/modules/core/keybinds/default_keybinds.lua [hunk 7]
```diff
@@ -13,9 +13,10 @@ module.public = {
                 n = { -- Bind keys in normal mode
 
                     -- Keys for managing TODO items and setting their states
-                    { "gtd", "core.norg.qol.todo_items.todo.task_done" },
                     { "gtu", "core.norg.qol.todo_items.todo.task_undone" },
                     { "gtp", "core.norg.qol.todo_items.todo.task_pending" },
+                    { "gtd", "core.norg.qol.todo_items.todo.task_done" },
+                    { "gtw", "core.norg.qol.todo_items.todo.task_wont_complete" },
```
Comment (mrossinek) [inline lua/neorg/modules/core/keybinds/default_keybinds.lua:19]: Do you also want to provide defaults for cancelling etc.?

File: queries/norg/highlights.scm [hunk 8]
```diff
@@ -160,3 +318,16 @@
...
+
+; Markup
+(bold) @NeorgMarkupBold
+(italic) @NeorgMarkupItalic
+(strikethrough) @NeorgMarkupStrikethrough
+(underline) @NeorgMarkupUnderline
+(spoiler) @NeorgMarkupSpoiler
+(verbatim) @NeorgMarkupVerbatim
+(superscript) @NeorgMarkupSuperscript
+(subscript) @NeorgMarkupSubscript
```
Comment (mrossinek) [inline queries/norg/highlights.scm:330]: You're defining these here but you don't set any highlights for them up top. I have some suggestions over in #117 which I will copy here:
```lua
        Spoiler = "+TSDanger",
        Subscript = "+TSString",
        Superscript = "+TSNumber",

        Math = "+TSMath",
        Variable = "+TSKeyword",
        Comment = "+TSComment",
 ```

File: lua/neorg/modules/core/norg/concealer/preset_basic.lua [hunk 9]
```diff
@@ -4,36 +4,30 @@ module.config.private.icon_preset_basic = {
     todo = {
         enabled = true,
```
Comment (mrossinek) [inline lua/neorg/modules/core/norg/concealer/preset_basic.lua:6]: I'm noticing just now that you're only updating the `basic` preset but not the others, although this is a change which needs to be applied to all of them.
This just strengthens my point that we need to remove this duplication to avoid errors such as this one in the future..
